ext/vscode/src/utils/process.ts

```diff
@@ -120,6 +154,28 @@ export async function spawnStreamAsync(
     onStderr?: (chunk: Buffer | string) => void,
     token?: vscode.CancellationToken): Promise<void> {
 
+    const useIntegratedAuth = vscode.workspace.getConfiguration('azure-dev').get<boolean>('auth.useIntegratedAuth', false);
+    let authServer: http.Server | undefined;
+
+    // TODO(ellismg): Do we want to add a guard here such that we only set up the auth server when the
+    // command implies that we are going to be running `azd`? In practice, I think we only ever
+    // use this function to run `azd` commands, but it might be nice to have the check to avoid spinning
+    // up the auth server and injecting the environment variables for a process that doesn't need them.
+    if (useIntegratedAuth) {
+        const { server, key } = await startAuthServer(new VsCodeAuthenticationCredential());
+        const serverEndpoint = `http://127.0.0.1:${(server.address() as AddressInfo).port}`;
```

Comment:
This line gets repeated several times; I'd consider returning it as part of the return result object. i.e. return `{ server, serverEndpoint, key }`

Reply:
Wonderful idea!  Done.